src/constructors.jl

```diff
@@ -103,3 +103,24 @@ end
 function Base.call{T, N}(::Type{NullableArray{T, N}})
     NullableArray(T, ntuple(i->0, N))
 end
+
+#----- Conversion from Array of Nullables #1 -----------------------------------#
+# Take an Array of Nullable objects and convert it into a NullableArray
+function Base.convert{S,T,N}(::Type{NullableArray{S,N}},in::Array{Nullable{T},N})
+   out = NullableArray{S,N}(Array(S,size(in)),Array(Bool,size(in)))
+   for i = 1:length(in)
+       (out.isnull[i] = in[i].isnull) ? nothing : out.values[i] = in[i].value
```

Comment:
Using an assignment and then an `if` block would be more natural. `nothing` is weird here, and the expression is complex. Also use `isnull(in[i])` rather than `in[i].isnull` and `get(in[i])` rather than `in[i].value`. Storing `in[i]` in an object might remove some overhead, not sure.